### ISSUE TITLE
Only a few nodes can access NodePort services through node_port

### ISSUE THREAD

**Issue by SmartLyu** [projectcalico/calico#2358]
Each node and pod are capable of running normally, and all nodes can be connected to the pod_IP of dns. The nginx service utilizes NodePort, but only a few nodes can have access to nginx through their ports.

## Expected Behavior

kubectl get svc
default       service/nginx                  NodePort    10.96.78.39    <none>        80:31781/TCP    63m   app=nginx,tier=backend

curl -v --connect-timeout 10 http://`hostname`:31781
172.16.0.216               : ok=2    changed=1    unreachable=0    failed=0
172.16.0.217               : ok=1    changed=0    unreachable=0    failed=1
172.16.0.218               : ok=1    changed=0    unreachable=0    failed=1
172.16.0.219               : ok=2    changed=1    unreachable=0    failed=0
172.16.0.220               : ok=2    changed=1    unreachable=0    failed=0
172.16.0.223               : ok=1    changed=0    unreachable=0    failed=1
172.16.0.224               : ok=2    changed=1    unreachable=0    failed=0
172.16.0.237               : ok=1    changed=0    unreachable=0    failed=1

As we can see, only a few nodes can be accessed through the specified Node_Port, which is apparently abnormal, thus I utilize the type of Node Port

All nodes can be accessed through the specified Node_Port, rather than just a few irregular nodes.

## Current Behavior

ssh  172.16.0.237   'tcpdump  -i tunl0'
tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
listening on tunl0, link-type RAW (Raw IP), capture size 262144 bytes

11:12:23.449434 IP JH-Api-JHCloudSH-Kubernate-Master1.38116 > 10.96.75.5.http: Flags [S], seq 532698513, win 43690, options [mss 65495,sackOK,TS val 5635621 ecr 0,nop,wscale 7], length 0
11:12:24.451578 IP JH-Api-JHCloudSH-Kubernate-Master1.38116 > 10.96.75.5.http: Flags [S], seq 532698513, win 43690, options [mss 65495,sackOK,TS val 5636624 ecr 0,nop,wscale 7], length 0
11:12:26.455585 IP JH-Api-JHCloudSH-Kubernate-Master1.38116 > 10.96.75.5.http: Flags [S], seq 532698513, win 43690, options [mss 65495,sackOK,TS val 5638628 ecr 0,nop,wscale 7], length 0

ssh  172.16.0.220   'tcpdump  -i tunl0'
tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
listening on tunl0, link-type RAW (Raw IP), capture size 65535 bytes

Through package capturing analysis, we can find that the data package was sent from the tunnel, but the other server system cannot receive it. In this case, there are only a few servers systems suffering from this condition. Oddly, it is unsuccessful from server A to server B, but from server B to server A can actually work properly. According to these operating problems, I think that it is not reasonable.

After a package is sent through a tunnel and forwarded through iptables, The node with specified IP should be able to make the response.

## Steps to Reproduce (for bugs)

1.
2.
3.
4.

## Your Environment

* Calico version
version:       3.4.0
Image:         quay.io/calico/cni:v3.4.0

* Orchestrator version (e.g. kubernetes, mesos, rkt):
kubernetes version:   1.13.1

* Operating System and version:
CentOS Linux release 7.3.1611

* Link to your project (optional):

**Comment by tmjd** [comment]
I believe it is possible to have a NodePort service that only "works" on hosts that have the service endpoint pods. Do the hosts that the NodePort works on also have one of the backend pods? Can you check the configuration of the node port service to make sure it should work on all hosts?

**Comment by caseydavenport** [comment]
Yeah, I'd first check if the service has `externalTrafficPolicy: Local` set
